tests/test_sequence_interpolator.py

```diff
@@ -0,0 +1,429 @@
+    ) as (timestamps, data, shift, seq_interp):
+        assert isinstance(
+            seq_interp, PhaseShiftedSequenceInterpolator
+        ), "Not a PhaseShiftedSequenceInterpolator"
+        seq_interp.interpolation_mode = "linear"
+
+        delta_t = 1.0 / sampling_rate
+        idx = slice(1, 11)
+        times = timestamps[idx] + 0.5 * delta_t
+
+        for sig_idx in range(data.shape[1]):
+            shift_offset = shift[sig_idx]
+            shifted_times = times + shift_offset
+
+            expected = np.zeros(len(shifted_times))
+            for i in range(len(shifted_times)):
+                t = shifted_times[i]
+
+                shifted_timestamps = timestamps + shift_offset
+                left_idx = np.searchsorted(shifted_timestamps, t, side="right") - 1
+                right_idx = left_idx + 1
+
+                if left_idx < 0 or right_idx >= len(timestamps):
+                    continue
+
+                t1, t2 = shifted_timestamps[left_idx], shifted_timestamps[right_idx]
+                y1, y2 = data[left_idx, sig_idx], data[right_idx, sig_idx]
+
+                expected[i] = y1 + ((t - t1) / (t2 - t1)) * (y2 - y1)
+
+            interp, valid = seq_interp.interpolate(times=shifted_times)
+
+            valid_indices = np.where(valid)[0]
+            if len(valid_indices) > 0:
+                assert np.allclose(
+                    interp[valid_indices, sig_idx], expected[valid_indices], atol=1e-6
+                ), f"Linear interpolation mismatch for signal {sig_idx}"
+
+
+@pytest.mark.parametrize("interpolation_mode", ["nearest_neighbor", "linear"])
+@pytest.mark.parametrize("keep_nans", [False, True])
+def test_interpolation_for_invalid_times(interpolation_mode, keep_nans):
+    end_time = 5.0
+    with sequence_data_and_interpolator(
+        data_kwargs=dict(
+            n_signals=10,
+            use_mem_mapped=True,
+            t_end=end_time,
+            sampling_rate=10.0,
+        ),
+        interp_kwargs=dict(keep_nans=keep_nans),
+    ) as (_, _, _, seq_interp):
+        assert isinstance(
+            seq_interp, SequenceInterpolator
+        ), "Interpolation object is not a SequenceInterpolator"
+        seq_interp.interpolation_mode = interpolation_mode
+
+        times = np.array([-5.0, -0.1, 0.1, 4.9, 5.0, 5.1, 10.0])
+        interp, valid = seq_interp.interpolate(times=times)
+        expected_valid = (
+            np.where((times >= 0.0) & (times <= end_time))[0]
+            if interpolation_mode == "nearest_neighbor"
+            else np.where((times >= 0.0) & (times < end_time))[0]
+        )
+        assert (
+            expected_valid == valid
+        ).all(), "Valid times does not match expected values"
+        assert interp.shape == (
+            3,
+            10,
+        ), f"Expected interp.shape == (2, 10), got {interp.shape}"
+
+
+@pytest.mark.parametrize("interpolation_mode", ["nearest_neighbor", "linear"])
+@pytest.mark.parametrize("end_time", [0.2, 1.0, 5.0])
```

Comment:
Could we add the test case when the valid should be empty? And the end time above the queried times,
like
```suggestion
@pytest.mark.parametrize("end_time", [0.091, 0.2, 1.0, 5.0, 12])
```
Also, what is the difference between 0.2 and 1.0 test cases? both should have length 1, or am I missing sth? (or is it exactly the point?)

Reply:
Adjusted test cases regarding end_time